client/client_test.go

```diff
@@ -1686,6 +1720,188 @@ func TestPublishTargetsDelgationSuccessNeedsToDownloadRoles(t *testing.T) {
 		[]string{"targets/a/b"})
 }
 
+// Ensure that two clients can publish delegations with two different keys and
+// the changes will not clobber each other.
+func TestPublishTargetsDelgationFromTwoRepos(t *testing.T) {
+	var tempDirs [2]string
+	for i := 0; i < 2; i++ {
+		tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+		assert.NoError(t, err, "failed to create a temporary directory: %s", err)
+		defer os.RemoveAll(tempBaseDir)
+		tempDirs[i] = tempBaseDir
+	}
+
+	gun := "docker.com/notary"
+	ts := fullTestServer(t)
+	defer ts.Close()
+
+	// this happens to be the client that creates the repo, but can also
+	// write a delegation
+	repo1, _ := initializeRepo(t, data.ECDSAKey, tempDirs[0], gun, ts.URL, true)
+	defer os.RemoveAll(repo1.baseDir)
+
+	// this is the second writable repo
+	repo2, err := NewNotaryRepository(tempDirs[1], gun, ts.URL,
+		http.DefaultTransport, passphraseRetriever)
+	assert.NoError(t, err, "error creating repository: %s", err)
+
+	// create keys for each repo
+	key1, err := repo1.CryptoService.Create("targets/a", data.ECDSAKey)
+	assert.NoError(t, err, "error creating delegation key")
+
+	// create a key on the delegated repo
+	key2, err := repo2.CryptoService.Create("targets/a", data.ECDSAKey)
+	assert.NoError(t, err, "error creating delegation key")
+
+	// delegation includes both keys
+	assert.NoError(t,
+		repo1.AddDelegation("targets/a", 1, []data.PublicKey{key1, key2}, []string{""}),
+		"error creating delegation")
+
+	assert.NoError(t, repo1.Publish())
+
+	// both repos add targets and publish
+	addTarget(t, repo1, "first", "../fixtures/root-ca.crt", "targets/a")
+	assert.NoError(t, repo1.Publish())
+	addTarget(t, repo2, "second", "../fixtures/root-ca.crt", "targets/a")
+	assert.NoError(t, repo2.Publish())
+
+	// first repo can publish again
+	addTarget(t, repo1, "third", "../fixtures/root-ca.crt", "targets/a")
+	assert.NoError(t, repo1.Publish())
+
+	// both should see both targets
+	for _, repo := range []*NotaryRepository{repo1, repo2} {
+		targets, err := repo.ListTargets()
+		assert.NoError(t, err)
+		assert.Len(t, targets, 3)
+
+		found := make(map[string]bool)
+		for _, t := range targets {
+			found[t.Name] = true
+		}
+
+		for _, targetName := range []string{"first", "second", "third"} {
+			_, ok := found[targetName]
+			assert.True(t, ok)
+		}
+	}
+}
+
+// A client who could publish before can no longer publish once the owner
+// revokes their delegation.
+func TestPublishRevokeDelgation(t *testing.T) {
```

Comment:
We shouldn't use the word revoke. It implies signing of something that stands on it's own. In this case, remove is more appropriate.